src/Computation.jl

```diff
@@ -0,0 +1,251 @@
+struct Model
+  alphaPrior::Vector{Float64} # concentration parameter for the symmetric Dirichlet prior on document topics
+  betaPrior::Vector{Float64} # concentration parameter for the symmetric Dirichlet prior on words
+  corpus::AbstractCorpus
+
+  # initialize an untrained model
+  Model(alphaPrior::Vector{Float64},
+        betaPrior::Vector{Float64},
+        corpus::AbstractCorpus) = begin
+    K = length(alphaPrior)
+    m = new(
+      alphaPrior,
+      betaPrior,
+      corpus)
+    return m
+  end
+
+  # initialize a trained model
+  Model(trainedModel::Model,
+        corpus::AbstractCorpus) = begin
+    m = new(
+      trainedModel.alphaPrior,
+      trainedModel.betaPrior,
+      corpus
+    )
+    return m
+  end
+end
+
+struct State
+  topics::Array{Float64,2}
+  topicSums::Vector{Float64}
+  docSums::Array{Float64,2}
+  assignments::Array{Array{Int64,1},1}
+  conditionals::Array{Array{Float64,2},1} # the p paramter for the word assignment (cat/multinom) variable
```

Comment:
This might not be an issue but in the past on large corpora where there's memory pressure keeping the conditions around winds up being the limiting factor. (As opposed to just keeping a temporary conditional for the word currently being sampled)